src/app/pages/shopping-list/shopping-edit/shopping-edit.component.ts

```diff
@@ -1,15 +1,25 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ElementRef, EventEmitter, OnInit, Output, ViewChild } from '@angular/core';
+import { Ingredient } from 'src/app/shared/ingredient.model';
 
 @Component({
   selector: 'app-shopping-edit',
   templateUrl: './shopping-edit.component.html',
   styleUrls: ['./shopping-edit.component.scss']
 })
 export class ShoppingEditComponent implements OnInit {
+ @ViewChild('nameInput') nameInputRef!: ElementRef;
+ @ViewChild('amountInput') amountInputRef!: ElementRef;
+ @Output() ingredientAdded = new EventEmitter<Ingredient>();
 
   constructor() { }
 
   ngOnInit(): void {
   }
 
+  onAddItem() {
+    const ingName = this.nameInputRef.nativeElement.value;
```

Comment:
Reactive forms can be used here (https://angular.io/guide/reactive-forms) with Required validator, please take a look on it